### DISCUSSION TITLE
v5.2.2

### DISCUSSION THREAD

**Discussion by niden** [phalcon/cphalcon#16362] [General]
<hr /><em>This discussion was created from the release <a href='https://github.com/phalcon/cphalcon/releases/tag/v5.2.2'>v5.2.2</a>.</em>

**Comment by s-ohnishi** [comment]
Phalcon can be compiled with PHP8.2, but not executed.
Because I get a lot of errors like:
(This is when the sample program INVO is executed)
```
Deprecated: Creation of dynamic property Phalcon\Mvc\View\Engine\Volt::$tag is deprecated in /var/www/var/cache/volt/%%var%%www%%themes%%invo%%index.volt .php on line 5
Deprecated: Creation of dynamic property Phalcon\Mvc\View\Engine\Volt::$session is deprecated in /var/www/var/cache/volt/%%var%%www%%themes%%invo%%layouts%% main.volt.php on line 12
Deprecated: Creation of dynamic property Phalcon\Mvc\View\Engine\Volt::$dispatcher is deprecated in /var/www/var/cache/volt/%%var%%www%%themes%%invo%%layouts%% main.volt.php on line 13
Deprecated: Creation of dynamic property Phalcon\Mvc\View\Engine\Volt::$flash is deprecated in /var/www/var/cache/volt/%%var%%www%%themes%%invo%%layouts%% main.volt.php on line 36
Deprecated: Use of "self" in callables is deprecated in /var/www/var/cache/volt/%%var%%www%%themes%%invo%%index%%index.volt.php on line 5
...
```

Regarding `dynamic property` in the first half, even if you can trick PHP by preparing a dummy __set() in a class that inherits Phalcon\Mvc\View\Engine\Volt, `Use of "self" in callables There is no other way but for ` to be supported by .zep, so general users cannot do anything.

With the release of Debian 12, the default is PHP 8.2, and even if you're using another distribution, you'll probably be thinking about migrating to PHP 8.2 later this year.
It became difficult to continue using Phalcon.
